lib/auth/auth_with_roles.go

```
@@ -3236,14 +3238,20 @@ func (a *ServerWithRoles) generateUserCerts(ctx context.Context, req proto.UserC
 	// This call bypasses RBAC check for users read on purpose.
 	// Users who are allowed to impersonate other users might not have
 	// permissions to read user data.
-	user, err := a.authServer.GetUser(ctx, req.Username, false)
+	impersonateUser, err := a.authServer.GetUser(ctx, req.Username, false)
 	if err != nil {
 		a.authServer.logger.DebugContext(ctx, "Could not impersonate user, the user could not be fetched from local store",
 			"error", err,
 			"user", req.Username,
 		)
 		return nil, trace.AccessDenied("access denied")
 	}
+	// TODO(greedy52) use a.authServer.GetUserOrLoginState when
+	// AccessChecker.CheckImpersonate supports services.UserState.
+	var user services.UserState = impersonateUser
+	if userState, err := a.authServer.GetUserLoginState(ctx, req.Username); err == nil {
```

Are we ok to ignore all errors here, or should we only ignore NotFound?